README.md

```diff
@@ -79,9 +79,28 @@ At the moment, your app is just getting data about one GitHub user. But what if
 Try and sketch out a plan!
 
 
-### Challenge 3: code coverage
+### Challenge 3: check code coverage of tests
+
+How could you test your app? Try and write some tests for the different parts of your application then check code coverage of those tests.
+
+Do you remember that you have been introduced to ```code coverge``` in research afternoon in week2 ?
+Basically, Code coverage is a measurement of how many lines/blocks/arcs of your code are executed while the automated tests are running.
+
+For example, a program with high code coverage, measured as a percentage, has had more of its source code executed during testing which suggests it has a lower chance of containing undetected software bugs compared to a program with low code coverage.
+
+**Why is test code coverage important?**
+
+This is because code coverage data gives an important insight of how effective our tests are, what parts of our source code are thoroughly executed to find out specific areas of code which are not executed by our tests.
+
+**Introduction code coverage tool**
+
+[Istanbul](https://www.npmjs.com/package/istanbul) and [NYC](https://www.npmjs.com/package/nyc) are widely-used tools for testing code coverage.
+You can install them as devdependencies by simply writing a command line ```npm install --save-dev istanbul``` or ```npm install --save-dev nyc``` on terminal. Nyc is Istanbul's state of the art command line interface. The original purpose of nyc was to allow a user to capture test coverage.
+These two started separately but have been merged.
+
```

Comment:
I don't know much about the difference between these NPM packages, but it sounds like they're quite similar? Would it be better to avoid confusion by sticking with one recommendation? In my experience, FAC people generally use istanbul...

Reply:
I think initially we used istanbul but then switched to nyc as it has full ES6 support?

Reply:
I thought nyc is a tool built on istanbul so that Istanbul and nyc are basically one set of code coverage tool now?
https://github.com/istanbuljs/nyc/issues/524
https://nodejs.libhunt.com/project/nyc/vs/istanbul

Reply:
@Heathercoraje can we just get them to use `nyc` rather than giving them choice?
if you're okay with that could you make the change to this section and push it

Reply:
@emilyb7 I don't know where @Heathercoraje but i've made a change. I get what you mean about the Code Coverage thing being kinda out of place, but that was the outcome of the the curriculum planning session and if it doesn't work it will hopefully serve to improve that process.
Let know your thoughts
Thanks

Reply:
@mantagen great thanks! I'm happy to merge as long as that's cool with the new process?

Reply:
merge at will